### PR TITLE
Bugfix/73 admin redirect save fix

### PR DESCRIPTION
**Pull request by ovidiul**
This PR fixes the UI relative path issues mentioned in #73 

Committed changes:

- the fix makes use of the home_url() function to compute the absolute path we should save in the database for backward compatibility, so users don't need to add the path prefix for multisites anymore in the UI
- added Follow link in backend for easier testing
- fixed the redirection  target URL, so when a relative path is provided, the home_url() is prefixed to it, added unit test as well

### CODE REVIEW

**Comment by ovidiul** [comment]
We might need to first merge PR #92 as this PR inherits the commits from it.

File: composer.json [hunk 1]
```diff
@@ -1,5 +1,4 @@
-{
-  "name": "automattic/wpcom-legacy-redirector",
+{                                                                                                                                                                                                                              "name": "automattic/wpcom-legacy-redirector",
```
Comment (GaryJones) [inline composer.json]: This excess whitespace seems to be back.

File: README.md [hunk 2]
```diff
@@ -21,6 +21,30 @@ Please contact us before using this plugin.
 - PHP 5.6+
 - WordPress 4.5+
 
+## Developement - Getting Started
```
Comment (GaryJones) [inline README.md]: This change in docs is unrelated (and should go into a `CONTRIBUTING.md` file anyway).

File: package.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{
```
Comment (GaryJones) [inline package.json]: This introduction of a package.json is unrelated.

File: .gitignore [hunk 4]
```diff
@@ -4,3 +4,4 @@
 /phpcs.xml
 /phpunit.xml
 /.phpunit.result.cache
+node_modules
```
Comment (GaryJones) [inline .gitignore]: This change is unrelated.

File: composer.json [hunk 5]
```diff
@@ -41,3 +40,4 @@
     ]
   }
 }
+
```
Comment (GaryJones) [inline composer.json]: The `composer.lock` has been included in this PR. Since `/composer.lock` is in the `develop` `.gitignore`, it suggests that you've not pulled `origin/develop` into your local `develop` branch before branching from it.

File: includes/class-wpcom-legacy-redirector-cli.php [hunk 6]
```diff
@@ -351,13 +351,15 @@ public function export_to_csv( $args, $assoc_args ) {
...
-			) );
+			$posts = get_posts(
+				array(
+					'posts_per_page'   => $posts_per_page,
+					'paged'            => $paged,
+					'post_type'        => WPCOM_Legacy_Redirector::POST_TYPE,
+					'post_status'      => 'any',
+					'suppress_filters' => 'false',
+				)
+			);
```
Comment (GaryJones) [inline includes/class-wpcom-legacy-redirector-cli.php]: This formatting change is unrelated.

File: includes/class-wpcom-legacy-redirector.php [hunk 7]
```diff
@@ -137,7 +137,7 @@ public static function wpcom_legacy_add_redirect_js( $hook ) {
 		}
 
 		wp_enqueue_script( 'wpcom-legacy-redirector', plugins_url( '/../js/admin-add-redirects.js', __FILE__ ), [], WPCOM_LEGACY_REDIRECTOR_VERSION, true );
-		wp_localize_script( 'wpcom-legacy-redirector', 'wpcomLegacyRedirector', array( 'siteurl' => get_option( 'siteurl' ) ) );
+		wp_localize_script( 'wpcom-legacy-redirector', 'wpcomLegacyRedirector', array( 'siteurl' => home_url() ) );
```
Comment (GaryJones) [inline includes/class-wpcom-legacy-redirector.php:111]: This change seems unrelated, but might be good to pull into a new issue + PR to explain why this change was warranted.

File: wpcom-legacy-redirector.php [hunk 8]
```diff
@@ -25,7 +25,7 @@
 define( 'WPCOM_LEGACY_REDIRECTOR_PLUGIN_NAME', 'WPCOM Legacy Redirector' );
 
 if ( defined( 'WP_CLI' ) && WP_CLI ) {
-	require __DIR__ . '/includes/class-wpcom-legacy-redirector-cli.php';
+	include __DIR__ . '/includes/class-wpcom-legacy-redirector-cli.php';
```
Comment (GaryJones) [inline wpcom-legacy-redirector.php]: This change is unrelated.

**Review by GaryJones — DISMISSED** [review]
Please remove the unrelated changes, and rebase your changes into one or two atomic commits.
